### PR TITLE
[Tests fixed!] Process DeviceCommands in simulated devices to ensure sync requests are correctly processed

### PR DESCRIPTION
**Pull request by Zee2**
## Overview
This PR will fix play-mode tests in batch mode (i.e., CI)!

image

- Made SimulatedControllers implement ExecuteCommand to receive Sync commands and report success, fixing the focus-lost-device-reset bug
- [x] Make SimulatedCamera implement the above ExecuteCommand method to report success on sync commands
- Changed our test harness to use the `InputTestFixture`
- Added `com.unity.inputsystem` to `testables`
- Manually registered XRI custom interactions/fallbackcomposites that were not registered in the custom input backend that the TestFixture uses, which was preventing us from using the `InputTestFixture`
- Fixed edge case in cursor magnetism script that was referencing colliders after they had been destroyed
- [x] Find out why roughly half of the tests continue to fail despite the input devices + interactors working correctly

**Breaking:** All references to "Simulated Camera" replaced with Simulated HMD for accuracy.

### CODE REVIEW

File: com.microsoft.mrtk.input/Interactors/Ray/MRTKRayInteractorVisual.cs [hunk 1]
```diff
@@ -304,7 +309,7 @@ public void LocateTargetHitPoint(SelectEnterEventArgs args)
                     targetLocalHitPoint = hitTargetTransform.InverseTransformPoint(raycastHit.Value.point);
                     targetLocalHitNormal = hitTargetTransform.InverseTransformPoint(raycastHit.Value.normal);
                 }
-
+
```
Comment (davidkline-ms) [inline com.microsoft.mrtk.input/Interactors/Ray/MRTKRayInteractorVisual.cs:312]: nit: what is this whitespace?

File: com.microsoft.mrtk.input/Simulation/InputSimulator.cs [hunk 2]
```diff
@@ -718,6 +719,32 @@ private string GetControlSetName(SimulatorControlScheme set)
...
+                // The state is maintained by XRDeviceSimulator, so no need for any change
+                // when focus is regained. Returning success instructs Input System to not
+                // reset the device.
+                result = InputDeviceCommand.GenericSuccess;
+                return true;
+            }
+
+            if (type == QueryCanRunInBackground.Type)
+            {
+                ((QueryCanRunInBackground*)commandPtr)->canRunInBackground = true;
```
Comment (davidkline-ms) [inline com.microsoft.mrtk.input/Simulation/InputSimulator.cs:739]: maybe add a comment here, it seems strange to see c++ syntax in c# code :smile:

File: com.microsoft.mrtk.input/Interactors/Ray/MRTKRayInteractorVisual.cs (hunk 1 above)
Comment (Zee2) [reply com.microsoft.mrtk.input/Interactors/Ray/MRTKRayInteractorVisual.cs]: Fixed

File: com.microsoft.mrtk.input/Simulation/InputSimulator.cs (hunk 2 above)
Comment (Zee2) [reply com.microsoft.mrtk.input/Simulation/InputSimulator.cs:744]: Commented

**Review by davidkline-ms — DISMISSED** [review]

**Review by keveleigh — APPROVED** [review]
